user_external/lib/imap.php

```diff
@@ -49,6 +49,32 @@ public function checkPassword($uid, $password) {
 			imap_close($mbox);
 			$uid = mb_strtolower($uid);
 			$this->storeUser($uid);
+			
+			if (preg_match('/^([^@]+)@(.+)\.[^.]+$/', $uid, $matches)) { 
```

Comment:
Maybe use `\OC::$server->getMailer()->validateMailAddress($uid)` first as it uses a library to verify in a more accurate manner.